thrust/testing/reduce_into.cu

```diff
@@ -189,8 +190,8 @@ void TestReduceIntoCountingIterator()
 
   ASSERT_LEQUAL(T(n), unittest::truncate_to_max_representable<T>(n));
 
-  thrust::counting_iterator<T, thrust::host_system_tag> h_first   = thrust::make_counting_iterator<T>(0);
-  thrust::counting_iterator<T, thrust::device_system_tag> d_first = thrust::make_counting_iterator<T>(0);
+  cuda::counting_iterator<T> h_first{0};
+  cuda::counting_iterator<T> d_first{0};
```

Comment:
Remark: here `thrust::reduce_into` deduces the system correctly, because there is a vector iterator as well from which to deduce the system.